### PR TITLE
feat: Add `InMemoryChatMessageStore` and `ChatMessageStore` 

### PR DESCRIPTION
**Pull request by vblagoje**
- Adds `ChatMessageStore` and `InMemoryChatMessageStore` as the main building block for ChatMessage storage
- Add unit tests for `InMemoryChatMessageStore` in `test/chat_message_stores/in_memory/test_chat_message_store.py`
- Omission of filters and top_k support is intentional and will be added in subsequent PRs
- As `ChatMessageStore` and `InMemoryChatMessageStore` are the main building blocks of chat message storage we'll add `ChatMessageRetriever` and `ChatMessageWriter` after this PR has been merged

### CODE REVIEW

File: haystack_experimental/chat_message_stores/in_memory/chat_message_store.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
```
Comment (shadeMe) [inline haystack_experimental/chat_message_stores/in_memory/chat_message_store.py:1]: Why do we need an extra `in_memory` submodule? I'd remove it.

File: haystack_experimental/chat_message_stores/types/protocol.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+from haystack import logging
+from haystack.dataclasses import ChatMessage
+
+# Ellipsis are needed for the type checker, it's safe to disable module-wide
+# pylint: disable=unnecessary-ellipsis
+
+logger = logging.getLogger(__name__)
+
+
+class ChatMessageStore(Protocol):
```
Comment (shadeMe) [inline haystack_experimental/chat_message_stores/types/protocol.py:16]: Why does this have to be a protocol instead of an `ABC`?

**Review by shadeMe — CHANGES_REQUESTED** [review]

File: haystack_experimental/chat_message_stores/in_memory/chat_message_store.py (hunk 1 above)
Comment (vblagoje) [reply haystack_experimental/chat_message_stores/in_memory/chat_message_store.py:1]: Yes, will do

File: haystack_experimental/chat_message_stores/types/protocol.py (hunk 2 above)
Comment (shadeMe) [reply haystack_experimental/chat_message_stores/types/protocol.py:16]: Let's just go with `ABC` for now.

**Review by shadeMe — APPROVED** [review]

**Pull request merged**